Event subtypes

wouterz opened the pull request:
Made event abstract, added first few sub types and added some tests

jessetilro commented:
I suggest you assign someone to review your code.

FlorisXll commented on src/main/java/nl/tudelft/pixelperfect/event/EventFactory.java:
```diff
@@ -6,51 +6,56 @@
-        return create(x);
-    }
+    int x = (int) Math.floor(Math.random() * 100) % 4;
+    return create(x);
```
I suggest doing (int) Math.random() * numberOfEvents instead of what you have now:
- the casting to int will floor the double.
- modulo 4 is not needed here?

jessetilro commented on src/main/java/nl/tudelft/pixelperfect/event/EventFactory.java:
```diff
@@ -0,0 +1,61 @@
+   * @return an Event.
+   */
+  public Event randomEvent() {
+    int x = (int) Math.floor(Math.random() * 100) % 4;
```
I recommend using the Random class rather than the Math class for generating random numbers.
http://stackoverflow.com/questions/738629/math-random-versus-random-nextintint

jessetilro commented on src/main/java/nl/tudelft/pixelperfect/event/EventFactory.java:
```diff
@@ -0,0 +1,61 @@
+   * @param x Defines which event is to be created.
+   * @return an Event.
+   */
+  public Event create(int x) {
```
For now this implementation is fine, it works. I do however suggest we refactor this in the future, for a switch statement is not the most elegant solution to this problem.

wouterz replied:
completely agree with this, was never intended to permanent, but couldnt think of a nicer way for now

jessetilro commented on src/test/java/nl/tudelft/pixelperfect/event/EventFactoryTest.java:
```diff
@@ -0,0 +1,71 @@
+  @Test
+  public void testCreateAsteroidEvent() {
+    Event e1 = factory.create(0);
+    assertTrue(e1 instanceof AsteroidField);
```
Not so sure whether instanceof is the best way to do this. You could for example use a method in the Event class (e.g. getType, returning a type identifier) that is implemented differently  by each subtype in order to verify them.

jessetilro commented on src/main/java/nl/tudelft/pixelperfect/event/PlasmaLeak.java:
```diff
@@ -0,0 +1,39 @@
+/**
+ * Created by woute on 5/10/2016.
+ */
+public class PlasmaLeak extends Event {
```
I suggest we make the names of Event subtypes contain the suffix 'Event', e.g. PlasmaLeakEvent.

jessetilro commented on src/test/java/nl/tudelft/pixelperfect/event/PlasmaLeakTest.java:
```diff
@@ -0,0 +1,16 @@
+        return new PlasmaLeak(1, "TestEvent", "An Event to test the Class.", 42, 42, 99.42);
+    }
+
+}
```
No newline at end of file, maybe fix this for I believe CheckStyle does not like this?

jessetilro commented on src/test/java/nl/tudelft/pixelperfect/event/HostileShipTest.java:
```diff
@@ -0,0 +1,16 @@
+   * Factory method for testing.
+   * @return HostileShip class
+   */
+  public HostileShip createEvent() {
```
No Override annotation here, as opposed to the factory method in for example the FireTest class?

FlorisXll commented:
LGTM, but you still have some CheckStyle warnings mainly in EvenFactory

wouterz commented:
Removed 1 unused import and changed 4 lines that were too long, still strongly suggest we remove that rule for checkstyle as it is worthless these days where we all have full hd monitors and it only decreases readability.
Other errors are not in my modified files and are/should have been fixed in other branches
